helper/genericdrive.h

```diff
@@ -17,30 +17,25 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-#ifndef RESTOREJOB_H
-#define RESTOREJOB_H
+#ifndef GENERICDRIVE_H
+#define GENERICDRIVE_H
 
 #include <QObject>
-#include <QProcess>
-#include <QTextStream>
+#include <QString>
 
-class RestoreJob : public QObject
-{
+class GenericDrive : public QObject {
```

Comment:
Well it might be smart to use composition for this. That would also make creating a fake drive. Would you be okay with that?
Because the way you describe your idea to improve this sounds like I have to duplicate code to me.

Reply:
Actually, most of the code would remain completely the same. It would mean adding a factory method (or a class), changing a few class names and maybe the type of the Drive variable in main.